.github/workflows/release.yml

```diff
@@ -33,7 +39,43 @@ jobs:
 
       - run: make -C .ci dist
 
-      - run: make -C .ci push-docker
+      - name: generate build provenance
+        uses: github-early-access/generate-build-provenance@main
+        with:
+          subject-path: "${{ github.workspace }}/build/aws/*.zip"
+
+      - id: setup-docker
+        name: Set up docker variables
+        run: |-
+          if [ "${{ startsWith(github.ref, 'refs/tags') }}" == "false" ] ; then
+            # for testing purposes
+            echo "tag=test" >> "${GITHUB_OUTPUT}"
+            echo "latest=test-latest" >> "${GITHUB_OUTPUT}"
+          else
+            # version without v prefix (e.g. 1.2.3)
+            echo "tag=${GITHUB_REF_NAME/v/}" >> "${GITHUB_OUTPUT}"
+            echo "latest=latest" >> "${GITHUB_OUTPUT}"
+          fi
+
+      # Use docker/build-push-action@v5.3.0 in conjunction with github-early-access/generate-build-provenance@main
+      - name: Build and Push Docker Image
+        id: push
```

Comment:
nit: I'd prefer that the `id`s used for these are changed to `docker-push` (rather than `push`) and perhaps `docker-setup` or `docker-vars`.   

- The `push` was a little hard for me to notice the `steps.push.outputs.digest` connection below.
- Having `docker-` as a common leading prefix seems helpful.